### PR TITLE
Jest tests feature

### PR DESCRIPTION
**Pull request by ViriCruz**
## This pull request contains the following features:
- capitalizeString function
- capitalizeString tests
- reverseString function
- reverseString tests
- ceasarCipher function
- ceasarCipher tests
- analize function
- analize tests

### CODE REVIEW

File: src/functions.test.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  expect(functions.cipher('helló world.', 1).ceasar()).toEqual('ifmmó xpsme.');
+});
+
+test('return encrypted alphabet', () => {
+  expect(functions.cipher('abcdefghijklmnopqrstuvwxyz', 1).ceasar()).toEqual('bcdefghijklmnopqrstuvwxyza');
+});
+
+test('key is greater than alphabet length', () => {
+  expect(functions.cipher('helló world.', 10).ceasar()).toEqual('rovvó gybvn.');
+});
```
Comment (mcrd25) [inline src/functions.test.js]: You missed a test that checked for keeping the same case and in so doing missed that your function does not handle this well and gives wrong results. 
For example, passing `functions.cipher('Z!', 1).ceasar()` should return `A!` but instead return `Z!` (Not that it simply returns original values of uppercase letters)
Please add the test and fix the original object.

See this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/testing_practice/caesar_requirement.png) from Odin description.

File: src/functions.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  return {
+    add,
+    subtract,
+    divide,
+    multiply,
+  };
+};
+
+const cipher = (string, shift) => {
+  const alphabet = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z'];
```
Comment (mcrd25) [inline src/functions.js]: What about uppercase letters?

Comment (Grifo89) [reply src/functions.js]: As you can review from the Odin project, the uppercase handler is not a requirement so, in this case, it is an optional feature.

File: src/functions.test.js (hunk 1 above)
Comment (Grifo89) [reply src/functions.test.js]: It is true, now it is working.

File: src/functions.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    });
+    return cipherString;
+  };
+  return {
+    ceasar,
+  };
+};
+
+const analyze = (arr) => {
+  const average = () => arr.reduce((a, i) => Math.ceil((a + i) / arr.length));
```
[the remark this answers is not in the document]
Comment (Grifo89) [reply src/functions.js]: Yes, that's right.

File: src/functions.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    });
+    return cipherString;
+  };
+  return {
+    ceasar,
+  };
+};
+
+const analize = (arr) => {
+  const average = () => Math.ceil((arr.reduce((a, i) => a + i, 0)) / arr.length);
```
Comment (MauricioRobayo) [inline src/functions.js]: I'm not sure about the ceiling of the result. The average of `[2, 3]` should be `2.5`, not `3`. Please give a second thought to this before merging.

File: src/functions.js (hunk 2 above)
Comment (mcrd25) [reply src/functions.js]: Oh, sorry about this, my mistake!

Comment (mcrd25) [reply src/functions.js]: Oh, sorry about this, my mistake!

**Pull request merged**